analysis/export_metrics_to_xlsx.py

```diff
@@ -0,0 +1,141 @@
+    sheets = [ws.title.split("_")[0] for ws in book.worksheets]
+
+    try:
+        # have to replace : since its illegal that a sheet name contains the character
+        instance_created_at = instances[0].created_at.replace(":", ".")
+
+        # creating the sheet name, adding suffix if other sheets with the same name exist
+        sheet_name = (
+            f"{instance_created_at}_1"
+            if instance_created_at not in sheets
+            else f"{instance_created_at}_{sheets.count(instance_created_at) + 1}"
+        )
+
+        # creating a multiindex object so that columns with the same name gets merged
+        columns = pd.MultiIndex.from_tuples(
+            column_tuples,
+            names=[
+                "Parameter",
+                "Parameter value",
+                "Policy",
+                "Shifts trained",
+                "Metrics",
+            ],
+        )
+
+        df = pd.DataFrame(
+            pd.concat(metrics_data, axis=1, ignore_index=True).to_numpy(),
+            columns=columns,
+        )  # concatenate all metrics dataframes to one dataframe
+
+        df.to_excel(
+            writer, sheet_name=sheet_name, startcol=1, startrow=1
+        )  # write dataframe to file
+
+        # TODO - add a sheet with globals hyper-parameters
+
+    finally:
+        writer.save()
+
+
+def add_metric_column(instance, metrics_data, column_tuples):
+    # making a dict of all metric variables and their values
+    metric_variables = instance.metrics.__dict__
+    # removing testing parameter name and value -> easier to have control over all metrics and add new one
+    parameter_name = metric_variables["testing_parameter_name"]
+    del metric_variables["testing_parameter_name"]
+    parameter_value = metric_variables["testing_parameter_value"]
+    del metric_variables["testing_parameter_value"]
+
+    for i, metric in enumerate(metric_variables.keys()):
+        # making the column name hierarchy to ensure that right name is associated with right list of values
+        column_tuples.append(
+            (
+                parameter_name.title(),
+                str(parameter_value),
+                instance.policy.__str__(),
+                str(instance.policy.value_function.shifts_trained)
+                if hasattr(instance.policy, "value_function")
+                else "N.A",
+                metric.replace("_", " ").title(),
+            )
+        )
+        # adding a dataframe of the metric data
+        # (have to be dataframe to be able to merge all metrics to one matrix later)
+        metrics_data.append(pd.DataFrame({i: metric_variables[metric]}))
+
+    return metrics_data, column_tuples
+
+
+def example_write_to_excel():
+    # TODO: rewrite so that a cached and trained world is used
+    POLICY = decision.EpsilonGreedyValueFunctionPolicy(
+        decision.value_functions.LinearValueFunction()
+    )
```

Comment:
Make it possible to run this file from the terminal with world object pickle as argument, @tordhaflan ?

Reply:
On it boss